scalding-core/src/main/scala/com/twitter/scalding/serialization/Writer.scala

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2015 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package com.twitter.scalding.serialization
+
+import java.io.OutputStream
+
+/**
+ * This is a specialized typeclass to make it easier to implement Serializations.
+ * The specialization *should* mean that there is no boxing and if the JIT
+ * does its work, Writer should compose well (via collections, Tuple2, Option, Either)
+ */
+trait Writer[@specialized(Boolean, Byte, Short, Int, Long, Float, Double) -T] {
+  def write(os: OutputStream, t: T): Unit
+}
+
+object Writer {
+  import JavaStreamEnrichments._
+
+  def write[@specialized(Boolean, Byte, Short, Int, Long, Float, Double) T](os: OutputStream,
+    t: T)(implicit w: Writer[T]): Unit =
+    w.write(os, t)
+  /*
+   * Instances below
+   */
+  implicit val unit: Writer[Unit] = new Writer[Unit] {
+    def write(os: OutputStream, u: Unit) = ()
+  }
+  implicit val boolean: Writer[Boolean] = new Writer[Boolean] {
+    def write(os: OutputStream, b: Boolean) = os.writeBoolean(b)
+  }
+  implicit val byte: Writer[Byte] = new Writer[Byte] {
+    def write(os: OutputStream, b: Byte) = os.write(b)
+  }
+  implicit val short: Writer[Short] = new Writer[Short] {
+    def write(os: OutputStream, s: Short) = os.writeShort(s)
+  }
+  implicit val int: Writer[Int] = new Writer[Int] {
+    def write(os: OutputStream, s: Int) = os.writeInt(s)
+  }
+  implicit val long: Writer[Long] = new Writer[Long] {
+    def write(os: OutputStream, s: Long) = os.writeLong(s)
+  }
+  implicit val float: Writer[Float] = new Writer[Float] {
+    def write(os: OutputStream, s: Float) = os.writeFloat(s)
+  }
+  implicit val double: Writer[Double] = new Writer[Double] {
+    def write(os: OutputStream, s: Double) = os.writeDouble(s)
+  }
+  implicit val string: Writer[String] = new Writer[String] {
```

Comment:
not sure I follow. You mean the code to read strings and write strings is far apart (so possibly easy to get out of sync)? If so, yeah, I see that point. Only testing seems like it can help us here.